### PR TITLE
update onceover

### PR DESCRIPTION
**Pull request by tuxmea**
This patch update sonceover and adds several onceover addons like codequality (lint), octocatalog diff and the onceover lookup.
octocatalog allows you to identify changes on servers by comparing the catalogs from branches
onceover lookup is similar to puppet lookup but can be done without puppet server and root access.

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -3,7 +3,7 @@
 FROM --platform=amd64 ruby:2.7.4-slim-buster as base
 
 ARG VCS_REF
-ARG GH_USER=puppetlabs
+ARG GH_USER=tuxmea
```
Comment (eputnam) [inline Dockerfile:6]:
```suggestion
ARG GH_USER=puppetlabs
```

File: Dockerfile [hunk 2]
```diff
@@ -22,7 +22,8 @@ ENV LC_ALL en_US.UTF-8
 RUN apt-get install -y apt-utils \
   && apt-get update -qq \
   && apt-get upgrade -y \
-  && apt-get install -y --no-install-recommends curl libxml2-dev libxslt1-dev g++ gcc git gnupg2 make openssh-client ruby-dev wget zlib1g-dev libldap-2.4-2 libldap-common libssl1.1 openssl \
+<<<<<<< HEAD
```
Comment (eputnam) [inline Dockerfile:25]: looks like maybe an artifact from some conflict resolution, unfortunately not valid Dockerfile syntax :)

**Review by eputnam — CHANGES_REQUESTED** [review]
Thanks for this @tuxmea! If you have time, it'd also be great to add any notes about what you're looking forward to in having this new version of onceover in your commit, i.e. was there something specific that motivated this change?

**Comment by tuxmea** [comment]
@eputnam sorry for the late reply. I did the changes and updated the mr description.

**Comment by eputnam** [comment]
was the `cmake` install on here before? what is requiring that?

**Comment by tuxmea** [comment]
@eputnam sorry for the late reply. I had to dig for myself:

onceover-octocatalog-diff depends on octocatalog-diff which depends on rugged, which needs cmake.

**Review by eputnam — APPROVED** [review]

**Pull request merged**